cmd/snap-bootstrap/initramfs_mounts_state.go

```diff
@@ -0,0 +1,144 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/boot"
+	"github.com/snapcore/snapd/seed"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/timings"
+)
+
+// initramfsMountsState helps tracking the state and progress
+// of the mounting driving process.
+type initramfsMountsState interface {
+	// Model returns the verified model from the seed (only for
+	// modes other than run).
+	Model() (*asserts.Model, error)
+
+	// RecoverySystemEssentialSnaps returns the verified essential
+	// snaps from the recoverySystem, if that is "" the implied one
+	// will be used (only for modes other than run).
```

Comment:
this is slightly hard to parse, what about

```suggestion
	// RecoverySystemEssentialSnaps returns the verified essential
	// snaps from the recoverySystem. If the recoverySystem is "" the
	// implied one will be used (only for modes other than run).
```